scilpy/io/image.py

```diff
@@ -23,3 +24,68 @@ def assert_same_resolution(images):
         aff_2, shape_2, _, _ = get_reference_info(i)
         if not (shape_1 == shape_2) and (aff_1 == aff_2).any():
             raise Exception("Images are not of the same resolution/affine")
+
+
+def get_data_as_mask(parser, in_img):
+    """
+    Get data as mask (force type np.uint8), check data type before casting.
+
+    Parameters
+    ----------
+    parser: argparse.ArgumentParser object
+        Parser.
+    in_img: nibabel.nifti1.Nifti1Image
+        Image
+
+    Return
+    ------
+    data: numpy.ndarray
+        Data (dtype : np.uint8).
+    """
+
+    curr_type = in_img.get_data_dtype().type
+    basename = os.path.basename(in_img.get_filename())
+    if np.issubdtype(curr_type, np.signedinteger) or np.issubdtype(curr_type, np.unsignedinteger) or np.issubdtype(curr_type, np.bool):
+        data = np.asanyarray(in_img.object).astype(np.uint8)
+        unique_vals = np.unique(data)
+        if len(unique_vals) == 2:
+            if np.all(unique_vals != np.array[0, 1]):
```

Comment:
A warning saying that the mask had two valy that were not  0 and 1 and converted to binary could be useful